### PR TITLE
chore: upgrade deps

### PR DESCRIPTION
**Pull request by fubuloubu**
fixes: #13

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -10,10 +10,19 @@
         "hypothesis>=6.2.0,<7.0",  # Strategy-based fuzzer
     ],
     "lint": [
-        "black>=22.6.0",  # auto-formatter and linter
-        "mypy==0.982",  # Static type analyzer
-        "flake8>=4.0.1",  # Style linter
+        "black>=22.12.0",  # auto-formatter and linter
+        "mypy>=0.991",  # Static type analyzer
+        "types-setuptools",  # Needed for mypy type shed
+        "flake8>=5.0.4",  # Style linter
```
Comment (unparalleled-js) [inline setup.py:16]:
```suggestion
        "flake8>=6.0.0",  # Style linter
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -10,18 +10,18 @@ repos:
...
-    rev: 4.0.1
+-   repo: https://github.com/pycqa/flake8
+    rev: 6.0.0
     hooks:
     -   id: flake8
 
 -   repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.971
+    rev: v0.991
     hooks:
```
Comment (unparalleled-js) [inline .pre-commit-config.yaml:25]: Can add the types-setuptools here

File: setup.py [hunk 3]
```diff
@@ -10,10 +10,19 @@
         "hypothesis>=6.2.0,<7.0",  # Strategy-based fuzzer
```
Comment (unparalleled-js) [inline setup.py:10]: The type ignore at the top of this file should no longer be needed now that we have `types-setuptools`.

File: setup.py [hunk 4]
```diff
@@ -10,10 +10,19 @@
...
     "lint": [
-        "black>=22.6.0",  # auto-formatter and linter
-        "mypy==0.982",  # Static type analyzer
-        "flake8>=4.0.1",  # Style linter
+        "black>=22.12.0",  # auto-formatter and linter
+        "mypy>=0.991",  # Static type analyzer
+        "types-setuptools",  # Needed for mypy type shed
+        "flake8>=5.0.4",  # Style linter
         "isort>=5.10.1",  # Import sorting linter
+        "mdformat>=0.7.16",  # Auto-formatter for markdown
```
Comment (unparalleled-js) [inline setup.py:18]: There are couple small changes to mdformat here: https://github.com/ApeWorX/project-template/pulls
Just adding the config file mostly.

**Review by helloibis — APPROVED** [review]

**Pull request merged**